File: frontend/src/types/prediction.ts
export type Detection = {
  xmin: number;
  ymin: number;
  xmax: number;
  ymax: number;
  score: number;
  label: string;
};

export type PredictionResponse = {
  tree_count: number;
  detections: Detection[];
  annotated_image_url: string;
};

File: backend/app/schemas/prediction.py
from pydantic import BaseModel


class Detection(BaseModel):
    xmin: float
    ymin: float
    xmax: float
    ymax: float
    score: float
    label: str


class PredictionResponse(BaseModel):
    tree_count: int
    detections: list[Detection]
    annotated_image_url: str
